test/cli/install/bun-install.test.ts

```diff
@@ -2027,92 +2027,6 @@ it("should handle ^0.0.2rc1 in dependencies", async () => {
   await access(join(package_dir, "bun.lockb"));
 });
 
-it("should handle ^0.0.2_pre3 in dependencies", async () => {
-  const urls: string[] = [];
-  setHandler(dummyRegistry(urls, { "0.0.2_pre3": { as: "0.0.2" } }));
-  await writeFile(
-    join(package_dir, "package.json"),
-    JSON.stringify({
-      name: "foo",
-      version: "0.0.1",
-      dependencies: {
-        bar: "^0.0.2_pre3+baz",
-      },
-    }),
-  );
-  const { stdout, stderr, exited } = spawn({
-    cmd: [bunExe(), "install"],
-    cwd: package_dir,
-    stdout: "pipe",
-    stdin: "pipe",
-    stderr: "pipe",
-    env,
-  });
-  const err = await new Response(stderr).text();
-  expect(err).toContain("Saved lockfile");
-  expect(err).not.toContain("error:");
-  const out = await new Response(stdout).text();
-  expect(out.replace(/\s*\[[0-9\.]+m?s\]\s*$/, "").split(/\r?\n/)).toEqual([
-    "",
-    "+ bar@0.0.2-_pre3",
-    "",
-    "1 package installed",
-  ]);
-  expect(await exited).toBe(0);
-  expect(urls.sort()).toEqual([`${root_url}/bar`, `${root_url}/bar-0.0.2.tgz`]);
-  expect(requested).toBe(2);
-  expect(await readdirSorted(join(package_dir, "node_modules"))).toEqual([".cache", "bar"]);
-  expect(await readdirSorted(join(package_dir, "node_modules", "bar"))).toEqual(["package.json"]);
-  expect(await file(join(package_dir, "node_modules", "bar", "package.json")).json()).toEqual({
-    name: "bar",
-    version: "0.0.2",
-  });
-  await access(join(package_dir, "bun.lockb"));
-});
-
-it("should handle ^0.0.2b_4+cafe_b0ba in dependencies", async () => {
```

Comment:
This looks like it was right before?

<img width="1261" alt="image" src="https://github.com/oven-sh/bun/assets/709451/05890a6a-beed-4b3f-a173-8c039cf091d2">

<img width="1287" alt="image" src="https://github.com/oven-sh/bun/assets/709451/67f72ad0-25ff-41ea-b60b-05ad59f9d0c1">

https://jubianchi.github.io/semver-check/#/%3E%3D0.0.2%20%3C0.0.3/0.0.2b_4%2Bcafe_b0ba

Reply:
oh i see, it's range versus version